### PR TITLE
*: use etcd for privilege update notification

### PR DESCRIPTION
**Pull request by tiancaiamao**
PD already have etcd embed, take full use of it.

### CODE REVIEW

**Comment by hanfei1991** [comment]
can you update vendor in another pr ?

**Comment by tiancaiamao** [comment]
I'll rebase it after https://github.com/pingcap/tidb/pull/3032

File: domain/notify/notify.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Copyright 2015 PingCAP, Inc.
```
Comment (zimulala) [inline domain/notify/notify.go]: s/2015/2017

File: session.go [hunk 2]
```diff
@@ -838,6 +843,18 @@ func BootstrapSession(store kv.Storage) (*domain.Domain, error) {
...
 	}
 	dom := sessionctx.GetDomain(se)
+	if ebd, ok := store.(etcdBackend); ok {
+		if addrs := ebd.EtcdAddrs(); addrs != nil {
+			cli, err := clientv3.New(clientv3.Config{
+				Endpoints:   addrs,
+				DialTimeout: 5 * time.Second,
+			})
+			if err != nil {
+				return dom, errors.Trace(err)
```
Comment (zimulala) [inline session.go]: s/dom/nil

File: domain/domain.go [hunk 3]
```diff
@@ -384,22 +400,44 @@ func (do *Domain) LoadPrivilegeLoop(ctx context.Context) error {
...
-		for {
-			select {
-			case <-ticker.C:
-				err := do.privHandle.Update()
-				if err != nil {
-					log.Error(errors.ErrorStack(err))
-				}
-			case <-do.exit:
-				return
+	// if etcd is available, notification i
```
Comment (zimulala) [inline domain/domain.go]: Please complete this comment.

**Comment by zimulala** [comment]
Do you forget to replace the `update` for `executeFlush`?

**Comment by coocood** [comment]
We don't need to make a `notify` package, notify should be a method of Domain.

**Comment by tiancaiamao** [comment]
PTAL @coocood @hanfei1991 @zimulala

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**